handlers/channel_handlers.go

```diff
@@ -107,6 +109,41 @@ func GetChannelIndexStats(c web.C, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func GetChannelHeaderFiles(c web.C, w http.ResponseWriter, r *http.Request) {
+	ch, found := findCachedChannel(c, "id")
+	if !found {
+		Render.JSON(w, http.StatusNotFound, map[string]string{"error": "channel not found"})
+		return
+	}
+
+	hashId, _ := ch.HashId()
+	query := QueryToMap(r.URL.Query())
+	if len(query) == 0 {
+		headerFiles := make(map[string]string)
+		for _, lang := range models.SupportedHeaderLanguages {
+			headerFiles[lang] = fmt.Sprintf("/channels/%s/header_files?language=%s", hashId, lang)
+		}
+		Render.JSON(w, http.StatusOK, headerFiles)
+		return
+	}
+
+	if len(query) > 1 || query["language"] == "" {
+		Render.JSON(w, http.StatusBadRequest, map[string]string{"error": "invalid query"})
+		return
+	}
+
+	filePath, fileName, err := models.FetchHeaderContentByChannel(ch, query["language"])
+
+	if err != nil {
+		Render.JSON(w, http.StatusBadRequest, map[string]string{"error": "header file not created"})
+		return
+	}
+
+	w.Header().Set("Content-Disposition", fmt.Sprintf("Attachment; filename=\"%s\"", fileName))
```

Comment:
It's shouldn't be an attachment.  Attachement is used for client upload to server.

You should be using `Content-Type: text/plain; charset="UTF-8"`